storage/Devices/PartitionImpl.cc

```diff
@@ -481,13 +518,57 @@ namespace storage
     Text
     Partition::Impl::do_set_id_text(Tense tense) const
     {
+	const PartitionTable* partition_table = get_partition_table();
+
 	string tmp = id_to_string(get_id());
 
-	if (tmp.empty())
-	    return sformat(_("Set id of partition %1$s to 0x%2$02X"), get_name().c_str(), get_id());
+	if (partition_table->get_impl().is_partition_id_value_meaningful())
```

Comment:
Not sure if I got why we make a difference between meaningful and not meaningful ids here. With meaningful but empty ids (surprising scenario, I would say) we don't fail but print a rather strange message "Set id of partition /dev/sda1 to (0x06)". With non-meaningful but empty messages we fail. Why is that worse?

Must be me (I'm a bit underpowered today), but the whole meaningfullnes topic does not have a clear meaning to me. :wink: